backend/corpora/lambdas/api/v1/common.py

```diff
@@ -35,3 +37,32 @@ def delete_dataset_common(db_session: Session, dataset: Dataset, token_info: dic
                 original.create_revision(dataset.collection.id)  # Restore the original dataset and S3 assets
             dataset.asset_deletion()  # Delete the S3 assets and database rows.
             dataset.delete()  # Delete the dataset row.
+
+
+def reshape_for_curation_api_and_is_allowed(collection, token_info, allow_access=False):
+    owner = collection["owner"]
+    if is_user_owner_or_allowed(token_info, owner):
+        collection["access_type"] = "WRITE"
+    elif not allow_access and collection["visibility"] == CollectionVisibility.PRIVATE:
```

Comment:
actually, to emphasize that it's a check for the user providing the uuid, what about renaming the flag something like `user_has_uuid`? Non-blocking suggestion, it just took me a little bit to decipher allow_access at first

Reply:
^ you can ignore the above if you like the idea of extracting this into two calls, which would obviate the flag entirely. otherwise, if you want to keep it one call, let me know what you think of this

Reply:
Sure, I'll think about that. I want to keep it as a flag because the collection id doesn't actually get used in the function itself.